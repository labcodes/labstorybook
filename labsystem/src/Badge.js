import React from "react";
import PropTypes from "prop-types";

export default class Badge extends React.Component {
  static propTypes = {
    type: PropTypes.oneOf(["floating-add", "harvest", "meta", "ponto", "star"])
      .isRequired,
    color: PropTypes.oneOf([
      "white",
      "black75",
      "mineral10",
      "mineral20",
      "mineral30",
      "mineral40",
      "mineral60",
      "mineral70",
      "mineral80",
      "mineral90",
      "teal40",
      "teal60",
      "teal70",
      "purple40",
      "purple60",
      "purple70",
    ]).isRequired,
    wrapperColor: PropTypes.oneOf([
      "white",
      "black75",
      "mineral10",
      "mineral20",
      "mineral30",
      "mineral40",
      "mineral60",
      "mineral70",
      "mineral80",
      "mineral90",
      "teal40",
      "teal60",
      "teal70",
      "purple40",
      "purple60",
      "purple70",
    ]).isRequired,
    size: PropTypes.oneOf(["small", "tiny"]),
  };

  static defaultProps = {
    size: undefined,
  };

  render() {
    const { type, color } = this.props;
    return <span className={`lab-badge ${color} ${type}`} />;
  }
}
