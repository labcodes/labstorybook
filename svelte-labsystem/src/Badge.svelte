<script>
  export let type = undefined;
  export let color = undefined;
  export let wrapperColor = undefined;

  let className = '';
  export { className as class };
</script>


<div class={`lab-badge__wrapper lab-badge__wrapper--${wrapperColor}`}>
  <span
    class={
      `lab-badge lab-badge--${color} lab-badge--${type}` +
      `${className ? ` lab-badge--${className}` : ""}`
    }
  />
</div>
