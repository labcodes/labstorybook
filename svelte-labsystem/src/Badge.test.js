import { render } from "@testing-library/svelte";
import Badge from "./Badge.svelte";

describe("Badge", () => {
  it("renders with base props", async () => {
    const { container } = render(Badge);
    expect(container).toMatchSnapshot();
  });

  it("renders with specific props", async () => {
    const { container } = render(Badge, {
      props: {
        class: "test",
        color: "teal-30",
        type: "harvest",
        wrapperColor: "mineral-70",
      },
    });

    expect(container).toMatchSnapshot();
  });
});
