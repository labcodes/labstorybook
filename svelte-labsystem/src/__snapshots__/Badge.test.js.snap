// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders with base props 1`] = `
<body>
  <div>
    <div
      class="lab-badge__wrapper lab-badge__wrapper--undefined"
    >
      <span
        class="lab-badge lab-badge--undefined lab-badge--undefined"
      />
    </div>
  </div>
</body>
`;

exports[`Badge renders with specific props 1`] = `
<body>
  <div>
    <div
      class="lab-badge__wrapper lab-badge__wrapper--mineral-70"
    >
      <span
        class="lab-badge lab-badge--teal-30 lab-badge--harvest lab-badge--test"
      />
    </div>
  </div>
</body>
`;
